src/services/registerService.js

```diff
@@ -0,0 +1,32 @@
+const bcrypt = require('bcrypt');
+const { putUser } = require("../repository/registerDAO");
+const { queryByUsername } = require('../repository/getUserDAO');
+
+async function register(req, res) {
+    const username = req.body.username;
+    let password = req.body.password;
+    const rounds = 10;
+    password = await bcrypt.hash(password, rounds);
+
+    const userExists = (await queryByUsername(username)).Count;
```

Comment:
This section of code is nice. I didn't realize dynamoDB queries had a Count attribute.

Reply:
Yes, it has `Count` which is the number of returned items, and `ScannedCount` which is how many items it searched through.